core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -234,7 +234,6 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
     }
     for (stageId <- jobData.stageIds) {
       stageIdToActiveJobIds.get(stageId).foreach { jobsUsingStage =>
-        jobsUsingStage.remove(jobEnd.jobId)
```

Comment:
I understand the issue, but now you're creating a memory leak, since these mappings will never be removed, right? Which will also cause the condition in the next line to never trigger, causing a memory leak in a different map too.